themes/qgrid/source/js/menu.js

```diff
@@ -15,41 +15,234 @@ function toggleVisibility(e) {
 	}
 }
 
-function filter(e) {
+function searchOnLoad() {
+	const active = document.querySelector('.active');
+	const search = getSearch();
+	if (active && search) {
+		pageTextSearch(search);
+		tagsSearch(search);
+		menuItemsSearch(search);
+	}
+}
+
+function searchOnHashChange() {
+	const search = getSearch();
+	updateMenuLinks(search);
+	pageTextSearch(search);
+	tagsSearch(search);
+	menuItemsSearch(search);
+}
+
+function onSearchChange(e) {
 	e.stopImmediatePropagation();
 	const input = e.target || e.srcElement;
-	const filter = input.value.toLowerCase();
+	const search = input.value.toLowerCase();
+	setSearch(search);
+}
+
+function setSearch(search) {
```

Comment:
move setSearch, getSearch to the bottom, try to put functions near each other that are similar